test/functional/wallet_taproot.py

```diff
@@ -511,16 +514,9 @@ def run_test(self):
 
         # Test 999-of-999
         self.do_test_k_of_n(999, 999)
-        # Test 1-of-1000, should not succeed
-        try:
-            self.do_test_k_of_n(1, 1000)
-        except JSONRPCException as e:
-            assert e.error["code"] == -5 and "Cannot have 1000 keys in multi_a; must have between 1 and 999 keys, inclusive" in e.error["message"]
-        # Test 999-of-1000, should not succeed
-        try:
-            self.do_test_k_of_n(999, 1000)
-        except JSONRPCException as e:
-            assert e.error["code"] == -5 and "Cannot have 1000 keys in multi_a; must have between 1 and 999 keys, inclusive" in e.error["message"]
+        # Test that all of the combinations 0-of-1000, 1-of-1000 and 999-of-1000 and 1000-of-1000 raise exception
+        for k in [0, 1, 999, 1000]:
+            assert_raises_rpc_error(-5, "Cannot have 1000 keys in multi_a; must have between 1 and 999 keys, inclusive", self.do_test_k_of_n, k, 1000)
```

Comment:
isn't it weird that the error message states "Cannot have 1000 keys" when k is not 1000?

Reply:
Maybe keep it simple and just test 1 of 1000.

Reply:
I guess that would do the trick, however I'm unsure if by doing so we modify the test to test the assertion, which seems a bit incomplete/unintuitive to me, rather than testing the edge cases in k-of-N wallets? 🤔 If you understand what I mean 😄 If you don't think it's an issue, I'm all for keeping it simple and just test 1-of-1000.

Reply:
Well the "edge" here is n = 999, but this test is varying `k`, not `n`.

Reply:
Ah you're right. I'll simplify the test to only do 1-of-1000.

Reply:
Would it make sense to test the other edge, 1-of-0?